lib/runners/processor/eslint.rb

```
@@ -191,12 +191,7 @@ def run_analyzer(target_dir, additional_options)
         *target_dir
       )
 
-      output_json =
-        if output_file.exist?
-          output_file.read.tap { |json| trace_writer.message json }
-        else
-          nil
-        end
+      output_json = File.file?(output_file) ? read_output_json(output_file) { nil } : nil
```

It would be useful `File.file?` is checked in `read_output_json` or `read_output_file`.

---

Good question. At first, I thought calling `File.file?` in `read_output_file`. But I noticed that it is not very meaningful. See the example below:

```ruby
def read_output_file(file)
  if File.file?(file)
    # reading file...
  else
    raise CustomFileNotFound # Should we add the custom error class instead of Errno::ENOENT?
    # Or should we return something?
  end
end
```

As you see the example above, a caller of `read_output_file` should raise an exception or receive a return value which expresses "File not found".

So, to keep it clear the implementation of `read_output_file`, I don't do `File.file?` in the method. That is, if needed, a caller of the method needs to call `File.file?` outside of the method.

What do you think about this decision?

---

It makes sense :+1: Yes, the complexity of the method grows with `File.file?`. I thought it's OK to return `nil`, but as you said, it might be meaningless.